### PR TITLE
Database module refactoring / repo reorganisation

### PR DESCRIPTION
**Pull request by sveinbjornt**
* Code in scraperdb.py and scraperinit.py is now in db/ subfolder (__init__, models, queries, setup)
* Shell scripts in repo root have been moved into scripts/ subfolder
* Some additional tests (now importing stuff in test code to check if imports fail)
* Some minor additions to Vocab
* Added Utanríkisráðuneyti to init_roots (again -- for some reason it wasn't part of the scraper-related changes the other day)

In terms of deployment, the paths to scripts in crontab will need updating. Also, we need to run ./scraper.py --init to create the Utanríkisráðuneyti root. For the topic vectors, a symlink to ../db should be created in the vectors folder.

### CODE REVIEW

File: config/Vocab.conf [hunk 1]
```diff
@@ -944,17 +944,21 @@ Kjósar-hreppur kk örn
 Suðurnesja-bær kk örn
 fiski-dauði kk
 Gaza-svæði hk örn
+Strass-borg kvk örn
 Trölla-skagi kk örn
 Voga-byggð kvk örn
 Árskógs-sandur kk örn
 Himalaja-fjöll hk örn # Ekki til með ákv. greini í BÍN
-Landmanna-laugar kvk örn # Orðmyndir eru "LandmannaLaugar" í BÍN, fjarlægt í BinErrata.conf
+Land-manna-laugar kvk örn # Orðmyndir eru "LandmannaLaugar" í BÍN, fjarlægt í BinErrata.conf
```
Comment (vthorsteinsson) [inline config/Vocab.conf]: The first hyphen is probably not needed.

File: db/__init__.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+from sqlalchemy import func as dbfunc
+
+from .models import Base
+
+# Allow client use of these exception without importing them from sqlalchemy
+IntegrityError = SqlIntegrityError
+DatabaseError = SqlError
+DataError = SqlDataError
+# Same for the desc() function
+desc = SqlDesc
```
Comment (vthorsteinsson) [inline db/__init__.py]: Is the previous code still needed now that this is in `__init__.py`? Isn't it simply enough to do `from sqlalchemy.exc import XXX as YYY`?

File: config/Vocab.conf [hunk 3]
```diff
@@ -955,6 +956,9 @@ streptó-kokkur kk
 Eurovision-keppni kvk
 Afríku-ríki hk
 Asíu-ríki hk
+Mið-Asíuríki hk
```
Comment (vthorsteinsson) [inline config/Vocab.conf]: This may need to be *Mið-Asíu-ríki*

File: .gitignore [hunk 4]
```diff
@@ -67,7 +67,7 @@ vectors/t
 vectors/test.py
 
 # Files that are symlinked into the vectors directory
-vectors/scraperdb.py
+vectors/db
```
Comment (vthorsteinsson) [inline .gitignore]: `/vectors/db/` ?

Comment (sveinbjornt) [reply .gitignore]: This will ignore a file or a folder.

File: db/queries.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This module wraos a number of SQLAlchemy SQL queries.
```
Comment (vthorsteinsson) [inline db/queries.py]: wraos -> wraps

File: main.py [hunk 6]
```diff
@@ -705,6 +695,9 @@ def analyze_api(version=1):
 def correct_api(version=1):
     """ Correct text manually entered by the user, i.e. not coming from an article.
         This is a lower level API used by the Greynir web front-end. """
+    if app.config["PRODUCTION"]:
+        return abort(403)
```
Comment (vthorsteinsson) [inline main.py]: Comment meaning of HTTP 403 code?

**Pull request merged**